innovation_sweet_spots/pipeline/pilot/investment_predictions/utils.py

```diff
@@ -0,0 +1,323 @@
+                for col in cols_to_loop
+            ]
+        )
+        .transpose()
+        .max(axis=1)
+    )
+    return cb_data
+
+
+def add_n_funding_rounds_in_window(
+    cb_data: pd.DataFrame, start_date: pd.DatetimeIndex, end_date: pd.DatetimeIndex
+) -> pd.DataFrame:
+    """Adds column for how many funding rounds occured in the date window
+
+    Args:
+        cb_data: Dataframe to add number funding rounds data to
+        start_date: Start date of the time window
+        end_date: End date of the time window
+
+    Returns:
+        Dataframe with column added for how many funding rounds occured
+        in the date window
+    """
+    cols_to_loop = cb_data.columns[cb_data.columns.str.contains("funding_round_date")]
+    cb_data["n_funding_rounds_in_window"] = (
+        pd.DataFrame(
+            [
+                pd.to_datetime(cb_data[col])
+                .between(start_date, end_date, inclusive="both")
+                .astype("int")
+                for col in cols_to_loop
+            ]
+        )
+        .transpose()
+        .sum(axis=1)
+    )
+    return cb_data
+
+
+def add_n_month_since_last_investment_in_window(
+    cb_data: pd.DataFrame, start_date: pd.DatetimeIndex, end_date: pd.DatetimeIndex
+) -> pd.DataFrame:
+    """Add column for number of months since last investment in the time window
+
+    Args:
+        cb_data: Dataframe to add number of months since last investment to
+        start_date: Start date of the time window
+        end_date: End date of the time window
+
+    Returns:
+        Dataframe with column added for number of months since last investment
+        in the time window
+    """
+    cols_to_loop = cb_data.columns[cb_data.columns.str.contains("funding_round_date")]
+    latest_investment_window_date = (
+        cb_data[cols_to_loop][
+            pd.DataFrame(
+                [
+                    pd.to_datetime(cb_data[col]).between(
+                        start_date, end_date, inclusive="both"
+                    )
+                    for col in cols_to_loop
+                ]
+            ).transpose()
+        ]
+        .astype("datetime64[ns]")
+        .max(axis=1)
+    )
+    cb_data["n_months_since_last_investment_in_window"] = (
+        ((end_date - latest_investment_window_date) / np.timedelta64(1, "M"))
+        .fillna(-1)
+        .astype(int)
+    )
+    return cb_data
+
+
+def add_n_month_since_founded(
+    cb_data: pd.DataFrame, end_date: pd.DatetimeIndex
+) -> pd.DataFrame:
+    """Add number of months since company was founded (where the
+    end of the time window date is the 'present' date)
+
+    Args:
+        cb_data: Dataframe to add the number of months since founded column to
+        end_date: End date of the time window to act as the 'present' date
+
+    Returns:
+        Dataframe with additional column for number of month since the company
+        was founded
+    """
+    cb_data["n_month_since_founded"] = (
```

Comment:
Super minor, but perhaps for consistency better to call this "n_months_since_founded"

Reply:
Good spot!